docs/DeveloperGuide.md

```diff
@@ -4,10 +4,154 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser
```

Comment:
Overall, developer guide gives me a clearer idea on what you guys are trying to do